### PR TITLE
 construct resource.Builder from kubeconfig flags

### PR DESCRIPTION
**Pull request by deads2k**
updates the resource.Builder to be constructed from the kubeconfig flags struct/interface that we now have.

@kubernetes/sig-cli-maintainers 
@juanvallejo @soltysh 

```release-note
the global flag --match-server-version is now global.  `kubectl version` will respect it.
```

### CODE REVIEW

**Comment by soltysh** [comment]
Depends on https://github.com/kubernetes/kubernetes/pull/63602 to be merged first

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -162,6 +162,21 @@ func NewBuilder(clientConfigFn ClientConfigFunc, restMapper meta.RESTMapper, cat
 	}
 }
 
+func NewBuilder(restClientGetter RESTClientGetter) *Builder {
+	restMapper, err := restClientGetter.ToRESTMapper()
```
Comment (soltysh) [inline pkg/kubectl/resource/builder.go]: One small nit is to use the error names like in the previous code, iow. `mapperError` and `clientError`, only for readability reasons :wink:

**Review by soltysh — APPROVED** [review]
One small nit that can be easily addressed in the rebase.

/lgtm

**Comment by deads2k** [comment]
/hold

I don't see the timeout probelm.  pushed a debug commit to find it.

**Comment by deads2k** [comment]
> /hold
> 
> I don't see the timeout probelm. pushed a debug commit to find it.

I still don't see it, but I can see that I need to make the kubectl restclientgetter shim first.

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**